pingcap/tidb-operator
# change the way to config tidb/tikv/pd in charts

### What problem does this PR solve? 
Do not provide config parameters for tidb/tikv/pd in values.yaml of charts any more to avoid the config mismatch between different releases of tidb/tikv/pd and also tidb-operator does not need to maintain the config templates for tidb/tikv/pd any more.
Instead, users should follow the config template in tidb/tikv/pd repo and provide customized configs by themselves.
### What is changed and how does it work?
Update the way to config tidb/tikv/pd in values.yaml of chart.
### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
    - Install TiDB cluster with new values.yaml and cluster runs normally, TiDB service is available to access.

Code changes

 - Has Helm charts change
 - Has documents change

Side effects

 - Breaking backward compatibility
    - Users should migrate the configs in values.yaml of previous chart releases to the new values.yaml of the new chart, otherwise, the configurations will be ignored when upgrading TiDB cluster with the new chart.

Related changes

 - Need to update the documentation

### Does this PR introduce a user-facing change?:
 
 ```release-note
Action required: Users should migrate the configs in values.yaml of previous chart releases to the new values.yaml of the new chart, otherwise, the configurations will be ignored when upgrading TiDB cluster with the new chart.
For example, configs in old values.yaml:
pd:
  logLevel: info
  maxStoreDownTime: 30m
  maxReplicas: 3
tikv:
  logLevel: info
  syncLog: true
  readpoolStorageConcurrency: 4
  readpoolCoprocessorConcurrency: 8
  storageSchedulerWorkerPoolSize: 4
tidb:
  logLevel: info
  preparedPlanCacheEnabled: false
  preparedPlanCacheCapacity: 100
  txnLocalLatchesEnabled: false
  txnLocalLatchesCapacity: "10240000"
  tokenLimit: "1000"
  memQuotaQuery: "34359738368"
  txnEntryCountLimit: "300000"
  txnTotalSizeLimit: "104857600"
  checkMb4ValueInUtf8: true
  treatOldVersionUtf8AsUtf8mb4: true
  lease: 45s
  maxProcs: 0

After migration, new values.yaml should be as below:
pd:
  config: |
    [log]
    level = "info"
    [schedule]
    max-store-down-time = "30m"
    [replication]
    max-replicas = 3
tikv:
  config: |
    log-level = "info"
    [raftstore]
    sync-log = true
    [readpool.storage]
    high-concurrency = 4
    normal-concurrency = 4
    low-concurrency = 4
    [readpool.coprocessor]
    high-concurrency = 8
    normal-concurrency = 8
    low-concurrency = 8
    [storage]
    scheduler-worker-pool-size = 4
tidb:
  config: |
    token-limit = 1000
    mem-quota-query = 34359738368
    check-mb4-value-in-utf8 = true
    treat-old-version-utf8-as-utf8mb4 = true
    lease = "45s"
    [log]
    level = "info"
    [prepared-plan-cache]
    enabled = false
    capacity = 100
    [txn-local-latches]
    enabled = false
    capacity = 10240000
    [performance]
    txn-entry-count-limit = 300000
    txn-total-size-limit = 104857600
    max-procs = 0

Please check [cluster configuration](https://pingcap.com/docs-cn/v3.0/reference/configuration/tidb-in-kubernetes/cluster-configuration/) for detailed configuration.
 ```
